pkg/hoist/hoist_launchable.go

```diff
@@ -22,37 +22,36 @@ import (
 type Fetcher func(string, string) error
 
 // A HoistLaunchable represents a particular install of a hoist artifact.
-type HoistLaunchable struct {
-	Location      string  // A URL where we can download the artifact from.
-	Id            string  // A unique identifier for this launchable, used when creating runit services
-	RunAs         string  // The user to assume when launching the executable
-	ConfigDir     string  // The value for chpst -e. See http://smarden.org/runit/chpst.8.html
-	FetchToFile   Fetcher // Callback that downloads the file from the remote location.
-	RootDir       string  // The root directory of the launchable, containing N:N>=1 installs.
-	Chpst         string  // The path to chpst
-	Contain       string  // The path to contain
-	ContainerType string  // The container type to pass to contain
+type Launchable struct {
+	Location     string         // A URL where we can download the artifact from.
+	Id           string         // A unique identifier for this launchable, used when creating runit services
+	RunAs        string         // The user to assume when launching the executable
+	ConfigDir    string         // The value for chpst -e. See http://smarden.org/runit/chpst.8.html
+	FetchToFile  Fetcher        // Callback that downloads the file from the remote location.
+	RootDir      string         // The root directory of the launchable, containing N:N>=1 installs.
+	Chpst        string         // The path to chpst
+	Cgexec       string         // The path to cgexec
+	CgroupConfig cgroups.Config // Cgroup parameters to use with cgexec
 }
 
 func DefaultFetcher() Fetcher {
 	return uri.URICopy
 }
 
-func (hoistLaunchable *HoistLaunchable) Halt(serviceBuilder *runit.ServiceBuilder, sv *runit.SV) error {
-
+func (hl *Launchable) Halt(serviceBuilder *runit.ServiceBuilder, sv *runit.SV) error {
 	// probably want to do something with output at some point
-	_, err := hoistLaunchable.Disable()
+	_, err := hl.disable()
 	if err != nil {
 		return err
 	}
 
 	// probably want to do something with output at some point
-	err = hoistLaunchable.Stop(serviceBuilder, sv)
+	err = hl.stop(serviceBuilder, sv)
```

Comment:
I suspect we'll need to change the access level for `disable` and `stop` back to public at some point.